pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -902,7 +902,14 @@ func (m *ManagerImpl) allocateContainerResources(pod *v1.Pod, container *v1.Cont
 
 		// Update internal cached podDevices state.
 		m.mutex.Lock()
-		m.podDevices.insert(podUID, contName, resource, allocDevices, resp.ContainerResponses[0])
+		allocDevicesWithNUMA := make(devicePerNUMA)
```

Comment:
> I might be nitpicking but the map building should probably be in a function :)

ok, I'll do. The PR for this patch was initially https://github.com/kubernetes/kubernetes/pull/93243.
So I'll update that PR, and if @fromanirh gives me access to this, this one too.

Reply:
@AlexeyPerevalov I'm fine giving your permissions, but in general is fine if you just fix your PR, I can rebase on top of yours and pick up your changes here.
I'm doing this because this PR depends on the data structure (and on some internal functions) changes you do on your PR, hence I replicated your commit here. Of course once your PR gets merged I'll remove my dupe.